qemu/tests/boot_order_check.py

```diff
@@ -74,25 +75,20 @@ def run(test, params, env):
                                          list_nic_addr[1][0],
                                          list_nic_addr[2][0])
 
-    error.context("Check the guest boot result", logging.info)
-    if bootorder_type == "type2":
-        session_serial = vm.wait_for_serial_login(timeout=timeout)
-        output = vm.serial_console.get_stripped_output()
-        session_serial.close()
-    else:
-        output = vm.serial_console.get_stripped_output()
-        utils_misc.wait_for(lambda: re.search("No bootable device.", output),
-                            timeout, 1)
+    error_context.context("Check the guest boot result", logging.info)
+    start = time.time()
+    while True:
         output = vm.serial_console.get_stripped_output()
-
-    # find and replace some ascii characters to non-ascii char,
-    # like as: '\b' (backspace)
-    if backspace_char:
-        data = re.sub(r".%s" % backspace_char, "", output)
-    else:
-        data = output
-    result = re.findall(boot_fail_infos, data, re.S | re.M | re.I)
-
+        # find and replace some ascii characters to non-ascii char,
+        # like as: '\b' (backspace)
+        if backspace_char:
+            data = re.sub(r".%s" % backspace_char, "", output)
+        else:
+            data = output
+        result = re.findall(boot_fail_infos, data, re.S | re.M | re.I)
```

Comment:
`re.S` and `re.M` should not be used at the same time, in this case, it should be `re.S`.

Reply:
Yes, you are right, modified it.

 (1/3) smp_4.256m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.RHEL.7.5.x86_64.myprovider.boot_order_check.bootorder0: PASS (224.37 s)
 (2/3) smp_4.256m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.RHEL.7.5.x86_64.myprovider.boot_order_check.bootorder1: PASS (223.58 s)
 (3/3) smp_4.256m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.RHEL.7.5.x86_64.myprovider.boot_order_check.bootorder2: PASS (227.44 s)